lib/internal/fs/promises.js

```diff
@@ -210,9 +218,31 @@ class FileHandle extends JSTransferable {
   [kDeserialize]({ handle }) {
     this[kHandle] = handle;
     this[kFd] = handle.fd;
+    this[kEmitter] = new EventEmitter();
+  }
+
+  [kRef]() {
+    this[kRefs]++;
+  }
+
+  [kUnref]() {
+    this[kRefs]--;
+    if (this[kRefs] === 0) {
+      this[kFd] = -1;
+      this[kHandle]
+        .close()
+        .then(this[kCloseResolve], this[kCloseReject]);
+    }
   }
 }
 
+for (const op of ['on', 'emit', 'once', 'off', 'addListener',
+                  'prependListener', 'removeListener', 'removeAllListeners',
+                  'listeners', 'rawListeners'])
```

Comment:
Do we maybe just to turn `FileHandle` into a subclass of `EventEmitter`?

```js
const EventEmitter = require('events');
const EventEmitterMixin = (Superclass) => {
  class MixedEventEmitter extends Superclass {
    constructor(...args) {
      super(...args);
      EventEmitter.call(this);
    }
  }
  const protoProps = Object.getOwnPropertyDescriptors(EventEmitter.prototype);
  delete protoProps.constructor;
  Object.defineProperties(MixedEventEmitter.prototype, protoProps);
  return MixedEventEmitter;
}

// ...

class FileHandle extends EventEmitterMixin(JSTransferable) {
  // ...
}
```

Reply:
This is a major decision. I didn't dare do it on my own :smile: 
I looked really hard in the code, I think this is the first time that a class that inherits from something must also be an EventEmitter.

Reply:
It plays nice with the `JSTransferable`, I tested it with WorkerThreads
Anyone else on this? Creating a new class generator  `EventEmitterMixin`?